url/src/lib.rs

```diff
@@ -2720,7 +2720,18 @@ impl Url {
                 _ => return Err(()),
             };
 
-            return file_url_segments_to_pathbuf(host, segments);
+            let estimated_capacity = self.as_str().len()
+                - if cfg!(target_os = "redox") {
```

Comment:
Can you make this a clamped subtract so that it doesn't wrap around in release when someone calls this with something other than `file` scheme URLs?

Reply:
Good catch. I think the logic needs to be a bit smarter here because urls could be like `a:/path`? I updated the code to at least be a fast path for `file://`-like urls and maybe in some weird cases like `a:/path` it might under allocate. Probalby not worth the effort to make it smarter than it is now.